lorawan-device/src/mac/session.rs

```diff
@@ -382,11 +386,20 @@ impl Session {
                         configuration.rx2_frequency = Some(freq);
                     }
 
+                    // RXParamSetupReq has its own acknowledgment mechanism, requiring
+                    // RXParamSetupAns with all uplinks until a Class A downlink is received
+                    // by the end-device.
                     let mut cmd = RXParamSetupAnsCreator::new();
                     cmd.set_rx1_data_rate_offset_ack(true)
                         .set_rx2_data_rate_ack(true)
                         .set_channel_ack(freq_ack);
-                    self.uplink.add_mac_command(cmd);
+
+                    self.uplink.rxparam_pending_ack(cmd.ack_get());
```

Comment:
there are also other commands that have the same ack mechanism. This is how I did it:
https://github.com/lucasgranberg/lorawan/blob/0c8446e037357f5949db7425c507948059080bae/src/mac/mod.rs#L824-L834

Reply:
Thanks, stole it from you :)